### DISCUSSION TITLE
Select witch window splits in split layout

### DISCUSSION THREAD

**Discussion by LamprosPitsillos** [qtile/qtile#3007] [Q&A] [answered]
Is  there a way to just select a window and put it on the split ? No Matches or anything complicated, just select it , press a key binding , and it becomes the split.

**Comment by elParaguayo** [comment]
Are you talking about the `Slice` layout?

The functionality isn't there at the moment but there's no reason it can't be done in a custom function.
``` python
@lazy.function
def window_to_slice(qtile):
    if not qtile.current_layout.name == "slice":
        return
    win = qtile.current_window
    slice = qtile.current_layout
    old_slice = slice._slice.window
    if old_slice:
        slice._slice.remove(old_slice)
        slice.fallback.add(old_slice)
    slice._slice.add(win)
    qtile.current_group.layout_all()

...

Key([mod], "s", window_to_slice(), desc="Move window to main slice"),
```
Completely untested so probably some errors in there but I'd do it something like that.

If this works, we could add it as a command to the layout so you could just call via `lazy.layout`.

**Reply by LamprosPitsillos** [reply]
Thanks! I will test it and if it proves to be stable please add it to lazy.layout, it would make the layout much easier to use

**Comment by LamprosPitsillos** [comment, marked as answer]
```python
checking qtile config file /home/inferno/.config/qtile/config.py                                                                                                                              
error: config failed to import: 'LazyCall' object is not callable                                                                                                                             
Stub: at line 1                                                                                                                                                                               
MypyFile:1(                                                                                                                                                                                   
  /tmp/tmp4s6hqhw0/config.pyi)                                                                                                                                                                
Runtime:                                                                                                                                                                                      
MISSING 
```


[Config gist](https://gist.github.com/LamprosPitsillos/b7e7f215869006bb05c630228c7d88db)

**Reply by elParaguayo** [reply]
OK.  You're not on latest. Try:
``` python
Key([mod], "s", window_to_slice, desc="Move window to main slice"),
```

**Reply by LamprosPitsillos** [reply]
Now it doesn't complain about , the config but still doesn't work.
Also i think current layout name should be the custom one the user sets.

**Reply by elParaguayo** [reply]
Yes, you'd need to change the name if you've set it in your config.

If it still doesn't work, I'd need to take a look later as I'm not on my laptop at the moment!

**Reply by LamprosPitsillos** [reply]
Yes i understand, thanks!

**Reply by elParaguayo** [reply]
Here you go:
``` python
@lazy.function
def window_to_slice(qtile):
    if not qtile.current_layout.name == "slice":
        return
    win = qtile.current_window
    slice = qtile.current_layout
    old_slice = slice._slice.window
    if old_slice:
        slice._slice.remove(old_slice)
        slice.fallback.add(old_slice)
        slice.layouts[old_slice] = slice.fallback
    slice.fallback.remove(win)
    slice._slice.add(win)
    slice.layouts[win] = slice._slice
    qtile.current_group.layout_all()
```

**Reply by LamprosPitsillos** [reply]
Thank you!! I will use it almost daily, hopefully this can become default functionality in the future.
Now the only thing remaining is being able for `lazy.layout.left() | right()` to change to the split but i think i saw a pull request or issue about it some time ago.

**Comment by LamprosPitsillos** [comment]
Hello , this stopped working a while ago... Is there a way for me to debug it? I really need this feature, thanks!
```
❯ qtile --version
0.22.1
```

**Reply by elParaguayo** [reply]
I've got an idea why it might not work but I thought that was a change after v0.22.1.

Are you getting any errors in the log?

**Reply by LamprosPitsillos** [reply]
Thanks for the quick response,
Nope, i don't get any errors in my logs.

**Reply by elParaguayo** [reply]
OK. It's a long shot (given you don't have any errors) but can you try running this command when the slice layout is the current layout:
```qtile cmd-obj -o layout -f eval -a "self._slice.add"```

**Reply by LamprosPitsillos** [reply]
```❯ qtile cmd-obj -o layout -f eval -a "self._slice.add"
(True,
 '<bound method Single.add of <libqtile.layout.slice.Single object at '
 '0x7f378a22dff0>>')
```
With nothing changing

**Reply by elParaguayo** [reply]
Didn't expect it to change anything. That just means what I thought may the issue, isn't!

Maybe I should just look at adding this to the layout and I can then debug it properly.

**Reply by LamprosPitsillos** [reply]
IMO without this feature the layout is kind of useless.
Would love to see it implemented natively !
Thanks either way.

**Reply by elParaguayo** [reply]
PR submitted: #4055
